docs/notebooks/dwi_simulated_gp.ipynb

```diff
@@ -0,0 +1,685 @@
+      "  warnings.warn(\n",
+      "/home/jhlegarreta/.virtualenvs/eddymotion/lib/python3.10/site-packages/sklearn/gaussian_process/kernels.py:455: ConvergenceWarning: The optimal value found for dimension 0 of parameter sigma_sq is close to the specified upper bound 10000.0. Increasing the bound and calling fit again may find a better value.\n",
+      "  warnings.warn(\n"
+     ]
+    }
+   ],
+   "execution_count": 11
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": "Now predict the signal on a set of diffusion-encoding gradient vector.  ",
+   "id": "351b0db081b0e890"
+  },
+  {
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-07-12T01:49:07.489499Z",
+     "start_time": "2024-07-12T01:49:07.468692Z"
+    }
+   },
+   "cell_type": "code",
+   "source": [
+    "X_qry = gtab.bvecs[1:]\n",
+    "prediction = gpfit.predict(X_qry)"
+   ],
+   "id": "ecfc20d16f567c91",
+   "outputs": [],
+   "execution_count": 12
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": "Check whether the hyperparameters of the kernel have been optimized",
+   "id": "c025c7d8a8c47b4c"
+  },
+  {
+   "metadata": {},
+   "cell_type": "code",
+   "source": "# gp_model.kernel_",
+   "id": "1ee8a8d2584ea19",
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": "Plot the training data and the predictions from the Gaussian process",
+   "id": "8913c3c3ef7d50f7"
+  },
+  {
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-07-12T01:49:22.789682Z",
+     "start_time": "2024-07-12T01:49:22.614716Z"
+    }
+   },
+   "cell_type": "code",
+   "source": [
+    "from matplotlib import pyplot as plt\n",
+    "%matplotlib inline\n",
+    "\n",
+    "# Exclude the b0s\n",
+    "s = dwi_data[voxel_idx[0], voxel_idx[1], voxel_idx[2], :][1:]\n",
+    "s_hat_mean = prediction\n",
+    "# s_hat_stddev = y_std[voxel_idx[0], voxel_idx[1], voxel_idx[2], :]\n",
+    "s_hat_stddev = np.ones_like(s_hat_mean)\n",
+    "x = np.asarray(range(len(gtab.bvals)))[1:]\n",
+    "\n",
+    "fig, ax = plt.subplots()\n",
+    "ax.plot(x, s_hat_mean, c=\"orange\", label=\"predicted\")\n",
+    "plt.fill_between(\n",
+    "    x.ravel(),\n",
+    "    s_hat_mean - 1.96 * s_hat_stddev,\n",
+    "    s_hat_mean + 1.96 * s_hat_stddev,\n",
+    "    alpha=0.5,\n",
+    "    color=\"orange\",\n",
+    "    label=r\"95% confidence interval\",\n",
+    ")\n",
+    "plt.scatter(x, s, c=\"b\", label=\"ground truth\")\n",
+    "ax.set_xlabel(\"bvec indices\")\n",
+    "ax.set_ylabel(\"signal\")\n",
+    "ax.legend()\n",
+    "plt.title(\"Gaussian process regression on dataset\")\n",
```

Comment:
The plot looks perfect because I am predicting on the very same data that I used for fitting.